src/main/java/seedu/application/logic/parser/SortCommandParser.java

```diff
@@ -0,0 +1,54 @@
+package seedu.application.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.application.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.application.logic.parser.CliSyntax.PREFIX_ORDER;
+import static seedu.application.logic.parser.CliSyntax.PREFIX_REVERSE;
+
+import java.util.Optional;
+
+import seedu.application.logic.commands.SortByCompanyCommand;
+import seedu.application.logic.commands.SortByDateCommand;
+import seedu.application.logic.commands.SortByPositionCommand;
+import seedu.application.logic.commands.SortCommand;
+import seedu.application.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new SortCommand object
+ */
+public class SortCommandParser implements Parser<SortCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the SortCommand
+     * and returns a SortCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public SortCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_ORDER,
+                PREFIX_REVERSE);
+
+        boolean shouldReverse = argMultimap.hasPrefix(PREFIX_REVERSE);
+        Optional<String> orderKeyword = argMultimap.getValue(PREFIX_ORDER);
+        if (orderKeyword.isEmpty()) {
+            // Sort by date by default if no order specified
+            return new SortByDateCommand(shouldReverse);
+        }
```

Comment:
I think when no order is specified and no reversal is specified, the list should be sort by reserve date by default

Reply:
Oh but then I'm worried it might be confusing since that would mean `sort` and `sort r/` both do the same thing.
Unless we also change the default date order to reverse chronological and require the `r/` prefix to change it to chronological, but that might be a bit unintuitive too imo

Reply:
Oh, make sense. Then just keep it this way first.